### ISSUE TITLE
Function calling support

### ISSUE THREAD

**Issue by controldev** [andrewyng/aisuite#55]
Is there any (plans for) function calling support? Without it, this has limted use in practical applications.

**Comment by rohitprasad15** [comment]
Tool calling support is planned and under development. It will be supported shortly.

**Comment by douglas125** [comment]
@rohitprasad15 I've been working on this multi-provider model calling topic in https://github.com/douglas125/SelfTestingGAT_LLM

maybe we could find a way to merge the 2 repositories in a way that I can focus on tools and aisuite can focus on the multi-providers?

I already added AWS Bedrock and function calling for many mainstream models, for example

**Comment by not-an-llm** [comment]
Will the initial implementation support parallel (multiple) tool/function calling over streamed responses across all providers?

**Comment by douglas125** [comment]
> Will the initial implementation support parallel (multiple) tool/function calling over streamed responses across all providers?

No, only one at a time at the moment. That's quite doable though
